ocs_ci/deployment/acm.py

```diff
@@ -0,0 +1,192 @@
+        # sequence number for the clusters from submariner perspective
+        # Used mainly to run submariner commands, for each cluster(except ACM hub) we will
+        # assign a seq number with 1 as primary and continue with subsequent numbers
+        self.cluster_seq = 1
+        # List of index to all the clusters which are participating in DR (except ACM)
+        # i.e index in the config.clusters list
+        self.dr_only_list = []
+
+    def deploy(self):
+        if self.source == "upstream":
+            self.deploy_upstream()
+        else:
+            self.deploy_downstream()
+
+    def deploy_upstream(self):
+        self.download_binary()
+        self.submariner_configure_upstream()
+
+    def deploy_downstream(self):
+        pass
+
+    def download_binary(self):
+        if self.source == "upstream":
+            # This script puts the platform specific binary in ~/.local/bin
+            # we need to move the subctl binary to ocs-ci/bin dir
+            download_cmd = "curl -Ls https://get.submariner.io | bash"
+            try:
+                run_cmd(download_cmd)
+            except CommandFailed:
+                logger.exception("Failed to download submariner binary")
+                raise
+
+            # Copy submariner from ~/.local/bin to ocs-ci/bin
+            # ~/.local/bin is the default path selected by submariner script
+            # TODO: create symlink
+            cp_cmd = f"cp ~/.local/bin/subctl {config.RUN['bin_dir']}"
+            ret = subprocess.run(shlex.split(cp_cmd))
+            try:
+                ret.check_returncode()
+            except subprocess.CalledProcessError:
+                logger.exception("Couldn't find subctl binary")
+                raise
+
+    def submariner_configure_upstream(self):
+        """
+        Deploy and Configure upstream submariner
+
+        Raises:
+            DRPrimaryNotFoundException: If there is no designated primary cluster found
+
+        """
+        gateway_node = self.get_default_gateway_node()
+        if self.designated_broker_cluster_index < 0:
+            raise DRPrimaryNotFoundException("Designated primary cluster not found")
+
+        # Deploy broker on designated cluster
+        # follow this config switch statement carefully to be mindful
+        # about the context with which we are performing the operations
+        config.switch_ctx(self.designated_broker_cluster_index)
+        logger.info(f"Switched context: {config.cluster_ctx.ENV_DATA['cluster_name']}")
+
+        deploy_broker_cmd = "deploy-broker"
+        try:
+            run_subctl_cmd(deploy_broker_cmd)
+        except CommandFailed:
+            logger.exception("Failed to deploy submariner broker")
+            raise
+
+        # Join all the clusters (except ACM cluster in case of hub deployment)
+        for cluster in config.clusters:
+            cluster_index = config.clusters.index(cluster)
```

Comment:
I see you are using this config.cluster.index(cluster) construction on a few places in the code.
What about to add the index to the cluster configuration directly when we are creating multi cluster configuration. I think that we don't change order so once we init it we should know the index and we could probably set it some of the section. Maybe you can have some method in configuration object for cluster which will hold its index value. WDYT?

Reply:
Done, now we are storing a map of Config object and its index as part of MulticlusterConfig object.